src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,265 @@
+    gss_OID desiredMech = &gss_mech_spnego_OID_desc;
+#endif
+
+    gss_buffer_desc inputToken{.length = UnsignedCast(inputLength), .value = inputBytes};
+    *outBufferHandle = new GssBuffer();
+
+    uint32_t majorStatus = gss_init_sec_context(minorStatus,
+                                                claimantCredHandle,
+                                                contextHandle,
+                                                targetName,
+                                                desiredMech,
+                                                reqFlags,
+                                                0,
+                                                GSS_C_NO_CHANNEL_BINDINGS,
+                                                &inputToken,
+                                                nullptr,
+                                                *outBufferHandle,
+                                                retFlags,
+                                                nullptr);
+
+    NetSecurityNative_HandleError(majorStatus, outBufferHandle, outTokenLength);
+    return majorStatus;
+}
+
+extern "C" uint32_t NetSecurityNative_ReleaseCred(uint32_t* minorStatus, GssCredId** credHandle)
+{
+    assert(minorStatus != nullptr);
+    assert(credHandle != nullptr);
+    return gss_release_cred(minorStatus, credHandle);
+}
+
+extern "C" uint32_t NetSecurityNative_ReleaseBuffer(uint32_t* minorStatus, GssBuffer* buffer)
+{
+    assert(minorStatus != nullptr);
+    assert(buffer != nullptr);
+    uint32_t status = gss_release_buffer(minorStatus, buffer);
+    delete buffer;
+    return status;
+}
+
+extern "C" void
+NetSecurityNative_CopyBuffer(GssBuffer* bufferHandle, uint8_t* bytes, uint32_t capacity, uint32_t offset)
+{
+    assert(bufferHandle != nullptr);
+    printf("capacity = %d, offset = %d, length = %zu \n", capacity, offset, bufferHandle->length);
+    assert(bufferHandle->length <= (capacity - offset));
+    memcpy(bytes + offset, bufferHandle->value, bufferHandle->length);
+}
+
+extern "C" uint32_t NetSecurityNative_ReleaseName(uint32_t* minorStatus, GssName** inputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+    return gss_release_name(minorStatus, inputName);
+}
+
+extern "C" uint32_t NetSecurityNative_Wrap(uint32_t* minorStatus,
+                                           GssCtxId* contextHandle,
+                                           int32_t isEncrypt,
+                                           uint8_t* inputBytes,
+                                           int32_t offset,
+                                           int32_t count,
+                                           GssBuffer** outBufferHandle,
+                                           uint32_t* outMsgLength)
+{
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(isEncrypt == 1 || isEncrypt == 0);
```

Comment:
I'll stop commenting on the asserts.  But please make sure that all of the arguments are appropriately validated throughout the code, e.g. pointers checked for nullptr.  If an argument can be null, it'd be good to comment on that so this doesn't come up in the future.